packer/windows/scripts/ec2-userdata.ps1

```diff
@@ -0,0 +1,32 @@
+<powershell>
+Start-Transcript
+
+write-output "Running User Data Script"
+write-host "(host) Running User Data Script"
+
+Set-ExecutionPolicy Unrestricted -Scope LocalMachine -Force -ErrorAction Ignore
+
+# Don't set this before Set-ExecutionPolicy as it throws an error
+$ErrorActionPreference = "stop"
+
+# Remove HTTP listener
+Remove-Item -Path WSMan:\Localhost\listener\listener* -Recurse
+
+$Cert = New-SelfSignedCertificate -CertstoreLocation Cert:\LocalMachine\My -DnsName "packer"
+New-Item -Path WSMan:\LocalHost\Listener -Transport HTTPS -Address * -CertificateThumbPrint $Cert.Thumbprint -Force
```

Comment:
Why do we need a self-hosted cert in the agent?

Reply:
This is not needed by the agent, but for Packer to be able to login into the instance for AMI provisioning. This script is referenced here: https://github.com/renderedtext/agent-aws-stack/pull/39/files#diff-5427e9e1ea449c944a6102e7e6f6939da3bebc288ec17ef41bff81e0ec47e93bR46